dbr quant: clean up logging format

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70200 dbr quant: start recording ops which are not quantizeable
* #70196 dbr quant: start reusing static quant module mappings
* #70128 dbr quant: better fix for bug with recursion on dequantize
* #70115 dbr quant: stop overridding tensor getters
* **#70114 dbr quant: clean up logging format**
* #70109 dbr quant: add test for qconfig_dict and methods

Summary:

This PR makes the debug logging for DBR quant be more useful
and easier to read.

New format looks like

```
DEBUG:auto_trace: fqn: _tf_ <function tanhshrink at 0x7fa4d02d4790> out torch.float32 end
```

This will be useful to speed up further work.

Test plan:

```
// run this with logging enabled, logs easier to read
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D33189545](https://our.internmc.facebook.com/intern/diff/D33189545)

vkuzo committed:
dbr quant: clean up logging format

Summary:

This PR makes the debug logging for DBR quant be more useful
and easier to read.

New format looks like

```
DEBUG:auto_trace: fqn: _tf_ <function tanhshrink at 0x7fa4d02d4790> out torch.float32 end
```

This will be useful to speed up further work.

Test plan:

```
// run this with logging enabled, logs easier to read
python test/test_quantization.py TestQuantizeDBR
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33189545).

jerryzh168 commented on torch/ao/quantization/_dbr/auto_trace.py:
```diff
@@ -87,17 +88,28 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
+            parent_module = cur_module
             if enable_logging:
-                logger.debug(f'__torch_function__ {str(func)} len_args {len(args)}')
+                is_obs_or_fq = isinstance(
```
I remember we have a helper function for this: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/quantize.py#L29
should probably move this to torch.ao.quantization.utils

jerryzh168 reviewed (approved):
looks good, had a nit comment

vkuzo committed:
Update on "dbr quant: clean up logging format"


Summary:

This PR makes the debug logging for DBR quant be more useful
and easier to read.

New format looks like

```
DEBUG:auto_trace: fqn: _tf_ <function tanhshrink at 0x7fa4d02d4790> out torch.float32 end
```

This will be useful to speed up further work.

Test plan:

```
// run this with logging enabled, logs easier to read
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D33189545](https://our.internmc.facebook.com/intern/diff/D33189545)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33189545).
